### DISCUSSION TITLE
(updated) errors in GHA, works fine locally...?

### DISCUSSION THREAD

**Discussion by lyricnz** [oxsecurity/megalinter#3306] [Q&A]
Created a PR with new megalinter workflow (basically stock, except replacing "master" with "develop")
https://github.com/Innablr/revolver/pull/36/files

But when it runs, it doesn't run any linters??  Seems to run OK locally using megalinter-runner.
https://github.com/Innablr/revolver/actions/runs/7606081048/job/20711303981

**Comment by lyricnz** [comment]
Changed to 
`VALIDATE_ALL_CODEBASE: true`
And
`uses: oxsecurity/megalinter/flavors/javascript@v7.8.0`

Still looks like it didn't run any checks?

**Reply by llaville** [reply]
Hello @lyricnz,

Perharps there is a contradiction between : 
- your workfile declaration : only `main` and `develop` branches are allowed here (see https://github.com/Innablr/revolver/actions/runs/7606081048/workflow#L90)
- and the branch you're pushing code `feature/megalinter` (see https://github.com/Innablr/revolver/actions/runs/7606081048/job/20711303981#step:4:52)

**Reply by lyricnz** [reply]
But the action appears to run, which means that GHA didn't have a problem with the trigger.

I modified the file-exclude setting to be a bit more careful about the regex:
```
FILTER_REGEX_EXCLUDE: '(\.github/workflows)'
```

...and it seems to be working now (or rather, I'm onto the next problem: how to make the GHA runner image work with typescript import-resolver).  I've got dozens of import-related errors

https://github.com/Innablr/revolver/actions/runs/7620990633/job/20756646888#step:4:146

```
  /github/workspace/drivers/ebs.ts
    1:53  error  Unable to resolve path to module '@aws-sdk/client-ec2'  import/no-unresolved
    2:26  error  Unable to resolve path to module 'luxon'                import/no-unresolved
```

Could this be something to do with the fact that we're using `.eslintrc.cjs` rather than the default?  (before my time)
🤯
